spec/ruby/core/struct/shared/inspect.rb

```diff
@@ -1,5 +1,28 @@
 describe :struct_inspect, shared: true do
+  it "returns a string representation showing members and values" do
+    car = StructClasses::Car.new('Ford', 'Ranger')
+    car.send(@method).should == '#<struct StructClasses::Car make="Ford", model="Ranger", year=nil>'
+  end
+
   it "returns a string representation without the class name for anonymous structs" do
     Struct.new(:a).new("").send(@method).should == '#<struct a="">'
   end
+
+  it "returns a string representation without the class name for structs nested in anonymous classes" do
+    obj = Object.new
+    obj.singleton_class.class_eval <<~DOC
```

Comment:
`Class.new` would be a bit simpler here, and mirror what is done below for modules, I'll simplify to that.

Reply:
Thanks. We wrote the spec looking closer to what IRB is doing, but changing that is fine.